File: 01-1.rb
#!/usr/bin/env ruby

lines = File.read('01.input').lines.map(&:strip)

left = []
right = []

lines.each do |line|
  l, r = line.split
  left << l.to_i
  right << r.to_i
end

left.sort!
right.sort!

puts left.zip(right).map { |l, r| (l - r).abs }.reduce &:+

File: 01-1.py
#!/usr/bin/env python

from functools import reduce

lines = [row.strip() for row in open('01.input').readlines()]

left = []
right = []

for line in lines:
    l, r = line.split()
    left.append(int(l))
    right.append(int(r))

left.sort()
right.sort()

print(reduce(lambda x, y: x + y, map(lambda lr: abs(lr[0] - lr[1]), zip(left, right))))
